Sources/Bow/Typeclasses/Divisible.swift

```diff
@@ -0,0 +1,12 @@
+/// Divisible extends Divide by providing an empty value.
+public protocol Divisible: Divide {
```

Comment:
@truizlop I have branched this off `452-add-divisible-type-class` as I needed `Divisible` to implement `Decidable`

Once that PR is in, I can rebase and Divisible and related files will go away.